early-results-test/src/main/java/com/hazelcast/jet/tests/earlyresults/EarlyResultsTest.java

```diff
@@ -112,21 +112,13 @@ protected void teardown(Throwable t) throws Exception {
         }
 
         void verify(KeyedWindowResult<String, Long> result) {
-            TickerWindow tickerWindow = tickerMap.computeIfAbsent(result.getKey(), TickerWindow::new);
-            // we have a result after the window advanced
-            // ignore if it is an early result, fail otherwise
-            if (result.start() < tickerWindow.start) {
-                logger.warning("Received a result after window advanced: " + result);
-                assertTrue(result.isEarly());
-                return;
-            }
-            //assertEquals(tickerWindow.start, result.start());
-            if (tickerWindow.start != result.start()) {
-                logger.severe("Did not receive the final result for the previous window:\n" +
-                        "Current result: " + result + "\n" +
-                        "Start index of tickerWindow=" + tickerWindow.start +
-                        ", start index of current result=" + result.start());
-                tickerWindow.resetTo(result.start());
+            TickerWindow tickerWindow = tickerMap.computeIfAbsent(result.getKey(), k -> new TickerWindow());
+
+            assertTrue("Received a result for the previous window: " + result,
+                    result.start() >= tickerWindow.start);
+
+            if (result.start() > tickerWindow.start) {
+                assertTrue("Received a final result for the next window: " + result, result.isEarly());
```

Comment:
Is this "Received a final result of a future window before receiving a result for the current window"?

Reply:
yes